src/algebra/big_operators/basic.lean

```diff
@@ -1208,6 +1208,15 @@ lemma prod_pow_boole [decidable_eq α] (s : finset α) (f : α → β) (a : α)
   (∏ x in s, (f x)^(ite (a = x) 1 0)) = ite (a ∈ s) (f a) 1 :=
 by simp
 
+lemma prod_dvd {S : finset α} (g1 g2 : α → β) (h : ∀ a ∈ S, g1 a ∣ g2 a) : S.prod g1 ∣ S.prod g2 :=
+begin
+  classical,
+  apply finset.induction_on' S, { simp },
```

Comment:
```suggestion
  induction S using finset.induction_on', { simp },
```

Reply:
Although I think you might not need the primed version here based on the `_` in  your intros below.

Reply:
I can get this to work with the unprimed version, although the proof then becomes a bit longer:
```
lemma prod_dvd_prod {S : finset α} (g1 g2 : α → β) (h : ∀ a ∈ S, g1 a ∣ g2 a) :
  S.prod g1 ∣ S.prod g2 :=
begin
  classical,
  induction S using finset.induction_on with a T haT h1, { simp },
  repeat {rw finset.prod_insert haT},
  apply mul_dvd_mul,
  { apply h, simp },
  { refine h1 (λ x hx, _), simp [h, mem_insert_of_mem hx] },
end
```

However, if I try using the primed version as you first suggested I immediately and consistently get `Server has stopped due to signal SIGSEGV` as soon as the compiler hits that line.  Any idea what's going wrong here, or should I just ask on Zulip?

Reply:
If you reliably get a sigsegv, then I'd push the broken version and ask on zulip what's going on.

Reply:
Hmm, just as mysteriously I'm no longer getting the `SIGSEGV` at all and I can't reproduce it.

Now `induction S using finset.induction_on'` just gives `induction tactic failed, failed to create new goal` (which is also strange, given that `apply finset.induction_on' S` does work).

Reply:
What's the advantage of `induction S using ...` rather than `apply ...`?  Is it just the ability to append `with ...` instead of the `intros` line?

Reply:
I can't find a way to get this proof shorter by `induction S using ...` than it currently is and I can't see any other advantage to using that style, so I'm going to leave this unchanged for now.  Sorry if I'm missing the point here.

Reply:
Well, the style guide wants
```suggestion
  apply finset.induction_on' S,
  { simp },
```
Which means my spelling is fewer lines that the version matching the style

Reply:
Really?  I understood the style guide to say that it's ok to immediately close a trivial goal on the same line, and it explicitly gives the example of `induction n, { simp }`.  A quick search turns up lots of examples of `induction ..., { simp...` on a single line (and several with `apply` or `refine`).

Reply:
Moreover I thought the aim in golfing proofs was to minimise the number of characters (since what we ultimately care about is the file size in bytes) rather than the number of lines.  Have I been misunderstanding this?

Reply:
I think we've always taken a pretty pragmatic approach to that aspect of the style guide.
Arguable the `induction .. with ..` shares more semantic information than the generic `apply`. But I wouldn't get hung up over this.

Reply:
Ok, thanks.  So is the proof ok as it stands (in this respect, at least)?